cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskPublisher.java

```diff
@@ -51,55 +48,34 @@
-                            PropertiesResolver propertiesResolver, CConfiguration cConf) {
+
+  public ScheduleTaskPublisher(Store store, MessagingService messagingService, CConfiguration cConf) {
     this.store = store;
     this.messagingService = messagingService;
-    this.propertiesResolver = propertiesResolver;
-    this.requirementsChecker = new RunConstraintsChecker(store);
     this.topicId = new TopicId(NamespaceId.SYSTEM.getNamespace(), cConf.get(Constants.Scheduler.TIME_EVENT_TOPIC));
   }
 
   /**
-   * Checks if all schedule requirements are satisfied,
-   * then publish notification to trigger the corresponding schedule.
+   * Publish notification for the triggered schedule
    *
-   * @param programId Program Id
-   * @param systemOverrides Arguments that would be supplied as system runtime arguments for the program.
-   * @param userOverrides Arguments to add to the user runtime arguments for the program.
+   * @param programId       {@link ProgramId} of the program to be run when the schedule is triggered
+   * @param retryCount       number of times of Quartz tries to re-fire trigger
+   * @param logicalStartTime the scheduled time the trigger fired for
    */
-  public void publishNotification(ProgramId programId, Map<String, String> systemOverrides,
-                                 Map<String, String> userOverrides) throws Exception {
-    Map<String, String> userArgs = Maps.newHashMap();
-    Map<String, String> systemArgs = Maps.newHashMap();
+  public void publishNotification(ProgramId programId, String scheduleName, int retryCount, long logicalStartTime)
+    throws Exception {
 
-    String scheduleName = systemOverrides.get(ProgramOptionConstants.SCHEDULE_NAME);
     ApplicationSpecification appSpec = store.getApplication(programId.getParent());
     if (appSpec == null) {
       throw new TaskExecutionException(String.format(UserMessages.getMessage(UserErrors.PROGRAM_NOT_FOUND), programId),
                                        false);
     }
 
-    ScheduleSpecification spec = appSpec.getSchedules().get(scheduleName);
-    if (!requirementsChecker.checkSatisfied(programId, spec.getSchedule())) {
-      return;
-    }
-
-    // Schedule properties are overridden by resolved preferences
-    userArgs.putAll(spec.getProperties());
-    userArgs.putAll(propertiesResolver.getUserProperties(programId.toId()));
-    userArgs.putAll(userOverrides);
-
-    systemArgs.putAll(propertiesResolver.getSystemProperties(programId.toId()));
-    systemArgs.putAll(systemOverrides);
-
     Map<String, String> properties = new HashMap<>();
-    properties.put("userArgs", GSON.toJson(userArgs));
-    properties.put("systemArgs", GSON.toJson(systemArgs));
     properties.put("programId", programId.toString());
-    properties.put("scheduleName", scheduleName);
+    properties.put(ProgramOptionConstants.SCHEDULE_NAME, scheduleName);
+    properties.put(ProgramOptionConstants.RETRY_COUNT, Integer.toString(retryCount));
```

Comment:
Not sure... the original TimeSchedule run programs with this as one of the runtime args. I just kept it in case it's useful. Haven't found any usage of ProgramOptionConstants.RETRY_COUNT in any other places though